pilot/pkg/model/service.go

```diff
@@ -582,3 +586,11 @@ func ParseLabelsString(s string) Labels {
 	}
 	return tag
 }
+
+// GetServiceAddressForProxy returns a Service's IP address specific to the cluster where the node resides
+func (s Service) GetServiceAddressForProxy(node *Proxy) string {
```

Comment:
Can you add few more comments here... Why, an example, etc - hard for people to understand.

Does this mean each service will have a different IP in different clusters ? Can it have a single IP instead ? There are some implications I'm not sure I understand from code alone.

Reply:
Different service IPs in different clusters are not required as long as the service IPs are local to the cluster. But if they are different, the change in this PR will ensure that the service IP local to the cluster is used to configure envoy. I explained this in one of the earlier comments.